nano/node/telemetry.hpp

```diff
@@ -149,51 +85,52 @@ class telemetry
 	 */
 	size_t telemetry_data_size ();
 
-	/*
-	 * Return the number of finished_single_requests elements
-	 */
-	size_t finished_single_requests_size ();
-
-	/*
-	 * Stop the telemetry processor
-	 */
-	void stop ();
-
 private:
+	class tag_endpoint
+	{
+	};
+	class tag_last_updated
+	{
+	};
+
 	nano::network & network;
 	nano::alarm & alarm;
 	nano::worker & worker;
 
+	std::atomic<bool> stopped{ false };
 	nano::network_params network_params;
-
-	class single_request_data
-	{
-	public:
-		std::shared_ptr<telemetry_impl> impl;
-		std::chrono::steady_clock::time_point last_updated{ std::chrono::steady_clock::now () };
-	};
+	bool disable_ongoing_requests;
 
 	std::mutex mutex;
-	/* Requests telemetry data from a random selection of peers */
-	std::shared_ptr<telemetry_impl> batch_request;
-	/* Any requests to specific individual peers is maintained here */
-	std::unordered_map<nano::endpoint, single_request_data> single_requests;
-	/* This holds data from single_requests after the cache is removed */
-	std::unordered_map<nano::endpoint, telemetry_data_time_pair> finished_single_requests;
-	bool stopped{ false };
-
-	void update_cleanup_data (nano::endpoint const & endpoint_a, nano::telemetry::single_request_data & single_request_data_a, bool is_new_a);
-	void ongoing_single_request_cleanup (nano::endpoint const & endpoint_a, nano::telemetry::single_request_data const & single_request_data_a);
-	void ongoing_req_all_peers ();
-
-	friend class node_telemetry_multiple_single_request_clearing_Test;
-	friend class node_telemetry_ongoing_requests_Test;
-	friend std::unique_ptr<container_info_component> collect_container_info (telemetry &, const std::string &);
+	// clang-format off
+	// This holds the last telemetry data received from peers or can be a placeholder awaiting the first response (check with awaiting_first_response ())
+	boost::multi_index_container<nano::telemetry_info,
+	boost::multi_index::indexed_by<
+		boost::multi_index::hashed_unique<boost::multi_index::tag<tag_endpoint>,
+			boost::multi_index::member<nano::telemetry_info, nano::endpoint, &nano::telemetry_info::endpoint>>,
+		boost::multi_index::ordered_non_unique<boost::multi_index::tag<tag_last_updated>,
+			boost::multi_index::member<nano::telemetry_info, std::chrono::steady_clock::time_point, &nano::telemetry_info::last_request>>>> recent_or_initial_request_telemetry_data;
```

Comment:
Minor: would be nice with an `namespace mi = boost::multi_index;` alias use here, like we do a few places already, for better readability.

Reply:
Done